air/tests/test_module/negative_tests/uncatchable_trace_related.rs

```diff
@@ -576,13 +576,102 @@ fn fold_pos_overflow() {
             subtrace_len: wrong_after_subtrace_len,
         },
     );
-    let resolved_fold = ResolvedFold::new(
-        hashmap![value_pos.into() => fold_lore],
-        (wrong_after_subtrace_len + 1) as usize,
-    );
+    let resolved_fold = ResolvedFold::new(hashmap![value_pos.into() => fold_lore], wrong_after_subtrace_len + 1);
     let expected_error = UncatchableError::TraceError {
         trace_error: StateFSMError(FoldPosOverflow(resolved_fold, before_subtrace_pos.into(), Previous)),
         instruction: String::from(String::from("fold $s i")),
     };
     assert!(check_error(&result, expected_error));
 }
+
+#[test]
+fn subtrace_len_overflow() {
+    let vm_peer_id_1 = "vm_peer_id_1";
+    let mut peer_vm_1 = create_avm(unit_call_service(), vm_peer_id_1);
+    let script = format!(
+        r#"
+        (par
+            (call "vm_peer_id_1" ("" "") [] $s)
+            (fold $s i
+                (call "vm_peer_id_2" ("" "") [] a)
+                (next i)
+            )
+        )
+    "#
+    );
+    let mut cid_state = ExecutionCidState::new();
+    let value_pos = 1;
+    let wrong_subtrace_len = u32::MAX;
```

Comment:
Can we write here `TraceLen::MAX`? I don't know if it would work.

Reply:
It works.